types/protocol/filter.go

```diff
@@ -0,0 +1,60 @@
+package protocol
+
+type FilterOperator int
+
+const (
+	Equal FilterOperator = iota
+	NotEqual
+	Lesser
+	LesserEqual
+	Greater
+	GreaterEqual
+)
+
+func (o FilterOperator) String() string {
+	switch o {
+	case Equal:
```

Comment:
Sure, why not offer it? :)
So either a NotFilter that can contain a filter or a Not for each of those.

Reply:
Hmmmm so a NotFilter we can't do (if we had 1.18 generics maybe we could), but maybe a `not bool` field that negates it.
But, why not just use the wildcards? These operators will probably never be implemented for anything but string filters. Would we have a need for them in numerical filters?

Reply:
Cool feature tbf @AsafEitani ! I suggest to insert it maybe in another PR and promote this, of course if everyone agrees :)

Reply:
Third option that can be done is adding helpers like the `protocol.EqualFilter` so something like:
`protocol.ContainsFilter(field string, vals ...string)` I think this would actually be better than new operators because it uses the already established wildcard logics internally and doesn't add **6** new operators that would only be used for one value type (which we can make explicit through the helper signature).

Reply:
Done